src/eduid_webapp/phone/views.py

```diff
@@ -0,0 +1,254 @@
+from flask import render_template, current_app
+
+from eduid_userdb.element import PrimaryElementViolation, DuplicateElementViolation
+from eduid_userdb.exceptions import UserOutOfSync
+from eduid_userdb.phone import PhoneNumber
+from eduid_common.api.decorators import require_dashboard_user, MarshalWith, UnmarshalWith
+from eduid_common.api.utils import save_dashboard_user
+from eduid_webapp.phone.schemas import PhoneListPayload, SimplePhoneSchema, PhoneSchema, PhoneResponseSchema
+from eduid_webapp.phone.schemas import VerificationCodeSchema
+from eduid_webapp.phone.verifications import send_verification_code
+
+
+phone_views = Blueprint('phone', __name__, url_prefix='', template_folder='templates')
+
+
+@phone_views.route('/all', methods=['GET'])
+@MarshalWith(PhoneResponseSchema)
+@require_dashboard_user
+def get_all_phones(user):
+    csrf_token = session.get_csrf_token()
+    phones = {'phones': user.phone_numbers.to_list_of_dicts(),
+              'csrf_token': csrf_token}
+    return PhoneListPayload().dump(phones).data
+
+
+@phone_views.route('/new', methods=['POST'])
+@UnmarshalWith(PhoneSchema)
+@MarshalWith(PhoneResponseSchema)
+@require_dashboard_user
+def post_phone(user, number, verified, primary, csrf_token):
+    if session.get_csrf_token() != csrf_token:
+        abort(400)
+
+    new_phone = PhoneNumber(number=number, application='dashboard',
+                            verified=False, primary=False)
+
+    try:
+        user.phone_numbers.add(new_phone)
+    except DuplicateElementViolation:
+        return {
+            '_status': 'error',
+            'error': {'form': 'phone_duplicated'}
+        }
+
+    try:
+        save_dashboard_user(user)
+    except UserOutOfSync:
+        return {
+            '_status': 'error',
+            'error': {'form': 'out_of_sync'}
+        }
+
+    send_verification_code(user, number)
+
+    phones = {'phones': user.phone_numbers.to_list_of_dicts()}
+    return PhoneListPayload().dump(phones).data
+
+
+@phone_views.route('/primary', methods=['POST'])
+@UnmarshalWith(SimplePhoneSchema)
+@MarshalWith(PhoneResponseSchema)
+@require_dashboard_user
+def post_primary(user, number, csrf_token):
+    if session.get_csrf_token() != csrf_token:
+        abort(400)
+
+    try:
+        phone_el = user.phone_numbers.find(number)
+    except IndexError:
+        return {
+            '_status': 'error',
+            'error': {'form': 'out_of_sync'}
+        }
+
+    if not phone_el.is_verified:
+        return {
+            '_status': 'error',
+            'error': {'form': 'phones.unconfirmed_number_not_primary'}
+        }
+
+    user.phone_numbers.primary = phone_el.number
+    try:
+        save_dashboard_user(user)
+    except UserOutOfSync:
+        return {
+            '_status': 'error',
+            'error': {'form': 'out_of_sync'}
+        }
+    phones = {'phones': user.phone_numbers.to_list_of_dicts()}
+    return PhoneListPayload().dump(phones).data
+
+
+def _steal_phone(number):
```

Comment:
I can't see that the phone number is being transfered to the new owner/user here - shouldn't that be a part of the same function?